boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +150,95 @@ impl Value {
 
         Self::object(object)
     }
+
+    /// Convert from a JSON value to a JS value
+    pub fn from_json(json: JSONValue, interpreter: &mut Interpreter) -> Self {
+        match json {
+            JSONValue::Number(v) => {
+                Self::rational(v.as_f64().expect("Could not convert value to f64"))
+            }
+            JSONValue::String(v) => Self::string(v),
+            JSONValue::Bool(v) => Self::boolean(v),
+            JSONValue::Array(vs) => {
+                let mut new_obj = Object::default();
+                for (idx, json) in vs.iter().enumerate() {
```

Comment:
I think we can consume the value and avoid the `json.clone()` below, by calling `into_iter()` instead of `iter()`. Is this possible? We would need to stract the length before, though, to use it below.

Reply:
:heavy_check_mark: